parsing/HACKING.adoc

```diff
@@ -25,6 +25,69 @@ target of the root Makefile to rebuild the compiler parser. See
 link:../Makefile.menhir[] for the details of the various
 Menhir-related targets and their use.
 
+In addition, it is recommended that you make sure to generate your modified
+parser with a version of Menhir which is at least as recent as the one
+used to generate the parser you are modifying (that is, to generate
+the files stored in the `boot/menhir` subdirectory). You should not
+use and older version of Menhir.
+
+If the version of Menhir you are using is more recent than the one
+used to generate the files under `boot/menhir`, then it is
+recommended that you do so in a dedicated commit which contains only
+this change and does not otherwise alter the parser.
```

Comment:
In this paragraph it is not clear what "do so" means. I think the edits made the whole thing less clear -- it's hard to keep overall consistency when making local changes.

I would give the following key informations, in this order:
- if your menhir version is different from the one used for the last promotion, `promote-menhir` will change the versioned copy of menhirLib
  (you can use  ¨git status" to see this)
- we only accept these changes if your menhir version is *more recent* than the one used by the compiler
  (and we prefer if it comes as a separate commit before your main change)
- here is how to check for the version used in the compiler sources
- here is how to check for your version, get opam-available versions, install a specific version